tests/test_task.py

```diff
@@ -1046,3 +1046,114 @@ def test_task_beatpos_tail(SR, HOP_LENGTH, SPARSE):
                            axis=0).astype(Y_pred.dtype)
     for i, (y1, y2) in enumerate(zip(Y_pred, Y_expected)):
         assert y1 == y2
+
+
+def test_task_key__encode_key_str(SPARSE):
+    # Checks the helper function which does key string to encoding
+    trans = pumpp.task.KeyTransformer(sparse=SPARSE)
+    
+    # Check A:minor
+    pitch_profile, tonic = trans._encode_key_str('A:minor')
+    assert np.all(pitch_profile == np.array([1, 0, 1, 0, 1, 1, 0, 1, 0, 1, 0, 1]))
+    if SPARSE:
+        assert tonic == 9
+    else:                            #   C #C  D bE  E  F #F  G bA  A bB  B  N
+        assert np.all(tonic == np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0]))
+
+    # Check 'N' for no key
+    pitch_profile, tonic = trans._encode_key_str('N')
+    assert np.all(pitch_profile == np.array(np.zeros(12, dtype=np.bool)))
+    # Or should the above the all 1's with np.ones()?
+    if SPARSE:
+        assert tonic == 12
+    else:
+        assert np.all(tonic == np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1]))
```

Comment:
it'd be good to have test coverage for different combinations (all combinations?) of tonic and mode

Reply:
I think I'll re think the unit test for encoding decoding mode. I'll refactor the encoding function out to separate tonic and profiles.

Reply:
So.... Ended up not touching the `_encode_key_string` function.
Added a couple more tests to test more combinations.
Not sure how to test all combinations efficiently without constructing all the `pitch_profile` vecs by hand.